tests/simple/memory.wast

```diff
@@ -1,33 +1,62 @@
-;; Test locals
-
-init_locals < i32 > 0 : < i32 > 0 : < i32 > 0 : .Stack
-
-i32.const 43
-set_local 0
-#assertLocal 0 < i32 > 43 "set_local"
-
-i32.const 55
-set_local 1
-get_local 1
-#assertTopStack < i32 > 55 "set_local stack"
-#assertLocal 1 < i32 > 55 "set_local"
-
-i32.const 67
-tee_local 2
-#assertTopStack < i32 > 67 "tee_local stack"
-#assertLocal 2 < i32 > 67 "tee_local local"
-
-;; Test globals
-
-init_global 0 < i32 > 0
-init_global 1 < i32 > 0
-
-i32.const 43
-set_global 0
-#assertGlobal 0 < i32 > 43 "set_global"
-
-i32.const 55
-set_global 1
-get_global 1
-#assertTopStack < i32 > 55 "set_global stack"
-#assertGlobal 1 < i32 > 55 "set_global"
+(module (memory 0)
+#assertMemory 0 pow16 .Bytes "Basic sanity"
+)
+
+(module (memory 0 0)
+#assertMemory 0 0 .Bytes "Basic sanity"
+)
+
+(module (memory 0 0) memory.size
+#assertTopStack < i32 > 0 "empty memory on stack"
+#assertMemory 0 0 .Bytes "empty memory"
+)
+
+(module (memory 15) memory.size
+#assertTopStack < i32 > 15 "memory size on stack"
+#assertMemory 15 pow16 .Bytes "empty memory"
+)
+
+
+
+(module
+  (memory 1)
+  (data (i32.const 0) "WASM")
+  #assertMemory 1 pow16 String2Bytes("WASM") "Bytes"
+  )
+(;
+
+(module
+  (memory 1)
+  (data (i32.const 0) "WASM")
+  (i32.load8_u (i32.const 0))
+  #assertTopStack < i32 > 87 "W in bytes" 
+)
+
+(module
+  (memory 1)
+  (data (i32.const 0) "WASM")
+  (i32.load8_u (i32.const 1))
+  #assertTopStack < i32 > 65 "A in bytes" 
+)
+
+(module
+  (memory 1)
+  (data (i32.const 0) "WASM")
+  (i32.load8_u (i32.const 2))
+  #assertTopStack < i32 > 83 "S in bytes" 
+)
+
+(module
+  (memory 1)
+  (data (i32.const 0) "WASM")
+  (i32.load8_u (i32.const 3))
+  #assertTopStack < i32 > 77 "M in bytes" 
+)
+
+(module
+  (memory 1)
+  (data (i32.const 0) "WASM")
+  (i32.load8_u (i32.const 4))
+  #assertTopStack < i32 > 0 "0 in bytes" 
+)
+;)
```

Comment:
I would prefer to be able to test memories without having them wrapped in a module if possible. This makes the language more "incremental", in the sense that someone can just pick it up and start writing as if at a repl without having to load up a module.

These hand-written tests should be about these atomic incremental actions that can be taken, whereas the actual wasm test-set can be used for full-blown module-based tests.

Reply:
Good point I'll fix that